src/main/java/com/sandvoxel/generitech/common/tileentities/power/TestPower.java

```diff
@@ -1,220 +1,55 @@
-                            GeneriTech.network.sendToServer(new PacketPower(T0transfer, pos.getX(), pos.getZ(), pos.getY() + -1));
-                        }
-                        this.container.takePower(T0transfer, false);
-                    } else {
-                        if (!worldObj.isRemote) {
-                            GeneriTech.network.sendToServer(new PacketPower(((ITeslaHolder) te).getStoredPower(), pos.getX(), pos.getZ(), pos.getY() + -1));
-                        }
-                        this.container.takePower(((ITeslaHolder) te).getStoredPower(), false);
-                    }
-                }
-            } else {
-                if (te2 instanceof TileEntityMachineBase && container.getStoredPower() != 0) {
-                    if (!worldObj.isRemote) {
-                        GeneriTech.network.sendToServer(new PacketPower(getTransfer(pos.down()), pos.getX(), pos.getZ(), pos.getY() + -1));
-                    }
-                    this.container.takePower(getTransfer(pos.down()), false);
-                }
-            }
-
-
-            if (te instanceof ITeslaHolder && te2 instanceof ITeslaHolder && te2 instanceof TileEntityMachineBase == false) {
-                long abbs = ((ITeslaHolder) te).getStoredPower() - ((ITeslaHolder) te2).getStoredPower();
-                //System.out.println(abbs);
-                if (abbs > 100) {
-                    if (!worldObj.isRemote) {
-                        GeneriTech.network.sendToServer(new PacketPower(T0transfer, pos.getX(), pos.getZ(), pos.getY() + -1));
-
-                    }
-                    this.container.takePower(T0transfer, false);
-                } else {
-                    if (te2 instanceof ITeslaConsumer && abbs != 0 && abbs / 2 != 0) {
-                        if (!worldObj.isRemote) {
-                            GeneriTech.network.sendToServer(new PacketPower(abbs / 2, pos.getX(), pos.getZ(), pos.getY() + -1));
-                        }
-                        this.container.takePower(abbs / 2, false);
-                        if (te2 instanceof ITeslaConsumer && abbs == 1) {
-                            if (!worldObj.isRemote) {
-                                GeneriTech.network.sendToServer(new PacketPower(1, pos.getX(), pos.getZ(), pos.getY() + -1));
-                            }
-                            this.container.takePower(1, false);
-                        }
-                    }
-                }
-
-            }
-
-        }
-
-
-        flag[0] = this.canConnectTo(worldIn, pos.up());
-        flag[1] = this.canConnectTo(worldIn, pos.down());
-        flag[2] = this.canConnectTo(worldIn, pos.north());
-        flag[3] = this.canConnectTo(worldIn, pos.south());
-        flag[4] = this.canConnectTo(worldIn, pos.east());
-        flag[5] = this.canConnectTo(worldIn, pos.west());
+        World wold = getWorld();
 
+        TeslaUtils.distributePowerToAllFaces(wold, pos, 100, false);
 
     }
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Rename this field "T0transfer" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AS00116)